zone/gm_commands/zsafecoords.cpp

```diff
@@ -2,25 +2,64 @@
 
 void command_zsafecoords(Client *c, const Seperator *sep)
 {
-	// modifys and resends zhdr packet
-	if (sep->arg[3][0] == 0) {
-		c->Message(Chat::White, "Usage: #zsafecoords <safe x> <safe y> <safe z>");
+	int arguments = sep->argnum;
+	if (
+		!arguments ||
+		!sep->IsNumber(1) ||
+		!sep->IsNumber(2) ||
+		!sep->IsNumber(3)
+	) {
+		c->Message(Chat::White, "Usage: #zsafecoords [X] [Y] [Z] [Heading] [Permanent (0 = False, 1 = True)]");
```

Comment:
Did you want to show defaults here?

Reply:
What defaults?

Reply:
Heading = current heading if blank, permanent = false. Was just checking if those needed to be identified.